controllers/clusteroperator_controller.go

```diff
@@ -144,38 +160,64 @@ func (r *CloudOperatorReconciler) Reconcile(ctx context.Context, _ ctrl.Request)
 	return ctrl.Result{}, nil
 }
 
-func (r *CloudOperatorReconciler) relatedObjects() []configv1.ObjectReference {
-	// TBD: Add an actual set of object references from getResources method
-	return []configv1.ObjectReference{
-		{Resource: "namespaces", Name: defaultManagementNamespace},
-		{Group: configv1.GroupName, Resource: "clusteroperators", Name: clusterOperatorName},
-		{Resource: "namespaces", Name: r.ManagedNamespace},
+func (r *CloudOperatorReconciler) sync(ctx context.Context, config operatorConfig) error {
+	// Deploy resources for platform
+	templates := getResources(config.Platform)
+	resources := fillConfigValues(config, templates)
+
+	updated, err := r.applyResources(ctx, resources)
+	if err != nil {
+		return err
 	}
+	if updated {
+		return r.statusProgressing(ctx)
+	}
+
+	return nil
 }
 
-func (r *CloudOperatorReconciler) sync(ctx context.Context, resources []client.Object) error {
+// applyResources will apply all resources as is to the cluster with
+// server-side apply patch and will enforce all the conflicts
+func (r *CloudOperatorReconciler) applyResources(ctx context.Context, resources []client.Object) (bool, error) {
+	updated := false
+
 	for _, resource := range resources {
-		if err := applyServerSide(ctx, r.Client, clusterOperatorName, resource); err != nil {
-			klog.Errorf("Unable to apply object %T '%s': %+v", resource, resource.GetName(), err)
-			return err
+		resourceExisting := resource.DeepCopyObject().(client.Object)
+		err := r.Get(ctx, client.ObjectKeyFromObject(resourceExisting), resourceExisting)
+		if errors.IsNotFound(err) {
+			klog.Infof("Resource %s %q needs to be created, updating...", resource.GetObjectKind().GroupVersionKind(), client.ObjectKeyFromObject(resource))
+			updated = true
+		} else if err != nil {
+			r.Recorder.Event(resource, corev1.EventTypeWarning, "Update failed", err.Error())
+			return false, err
+		}
+
+		resourceUpdated := resource.DeepCopyObject().(client.Object)
+		if err := r.Patch(ctx, resourceUpdated, client.Apply, client.ForceOwnership, client.FieldOwner(clusterOperatorName)); err != nil {
+			klog.Errorf("Unable to apply object %s '%s': %+v", resource.GetObjectKind().GroupVersionKind(), resource.GetName(), err)
+			r.Recorder.Event(resourceExisting, corev1.EventTypeWarning, "Update failed", err.Error())
+			return false, err
+		}
+		klog.V(2).Infof("Applied %s %q successfully", resource.GetObjectKind().GroupVersionKind(), client.ObjectKeyFromObject(resource))
+
+		if resourceExisting.GetGeneration() != resourceUpdated.GetGeneration() {
+			klog.Infof("Resource %s %q generation increased, updating...", resource.GetObjectKind().GroupVersionKind(), client.ObjectKeyFromObject(resource))
```

Comment:
Maybe change this message to smth like ""Resource %s %q updated.""? As far as I understand resource already updated there.

Reply:
Or specify somehow what is gonna update here.